### PR TITLE
Made issue #2434 improvement

### PR DESCRIPTION
**Pull request by jorgesumle**
issue URL: https://github.com/unknown-horizons/unknown-horizons/issues/2434

### CODE REVIEW

**Comment by LinuxDonald** [comment]
I will test that tomorrow :-)

**Comment by LinuxDonald** [comment]
@MasterofJOKers could you maybe do an code review here? I will test it tomorrow.

File: horizons/gui/modules/settings.py [hunk 1]
```diff
@@ -172,12 +175,15 @@ def apply_settings(self):
...
-			self.show_restart_popup()
-
 		self.hotkey_interface.save_settings()
 		self._settings.apply()
 		self._settings.save()
+
+		if restart_required:
+			if self.restart_promt():
+				horizons.globals.fife.engine.destroy()
+				os.execl(sys.executable, sys.executable, *sys.argv)
```
Comment (MasterofJOKers) [inline horizons/gui/modules/settings.py]: From reading the [documentation](https://docs.python.org/2/library/os.html#os.execl), `os.execv` should be used here.

**Comment by MasterofJOKers** [comment]
Other than that, it looks like it should work.

**Comment by jorgesumle** [comment]
os.execv works fine here. You mean with `os.execv(sys.executable, sys.argv)`?

**Comment by MasterofJOKers** [comment]
No, I mean
```python
args = [sys.executable] + sys.argv
os.execv(sys.executable, args)
```
The following part in the documentation looked like it shouldn't be used that way:
> The “l” variants are perhaps the easiest to work with if the number of parameters is fixed when the code is written;

Why would they write something like that with Python supporting `*args`? Maybe I'm interpreting too much into that sentence, but looking at the documentation, `os.execl()` seems to be the wrong choice - but since it seems to work, I'm not totally against using it. It just feels wrong to me.

**Comment by jorgesumle** [comment]
Yes, that should work too. In the end, os.execl calls os.execv so they are practically the same.

**Comment by MasterofJOKers** [comment]
Please don't edit your comments in a totally meaning-changing way.

**Pull request closed without merging**